docs/rst/tutorials/micro/custom_platforms/custom_platforms.rst

```diff
@@ -0,0 +1,341 @@
+
+.. code-block::
+
+    [TRIPLET PREFIX]-gcc -o microros_app.elf [COMPILER FLAGS] -I[MICROROS INCLUDE FOLDER] main.c libmicroros.a
+
+.. note::
+
+    Note that ``[COMPILER FLAGS]`` shall be the same when generating the micro-ROS Client library and when building the micro-ROS application.
+
+Makefile
+^^^^^^^^
+
+An example on how to integrate micro-ROS Client library in a Make build system can be found in the `micro-ROS app for TI Tiva C Series <https://github.com/micro-ROS/micro_ros_tivac_launchpad_app/blob/humble/Makefile>`_.
+
+CMake
+^^^^^
+
+An example on how to integrate micro-ROS Client library in a CMake build system can be found in the `micro-ROS example for Mbed RTOS <https://github.com/micro-ROS/micro_ros_mbed/blob/humble/CMakeLists.txt>`_.
+
+
+micro-ROS system dependencies
+-----------------------------
+
+There are three points where micro-ROS Client library needs to use functionality of the target platform beyond the ``libc`` implementation:
+
+- when obtaining a time reference
+- when configuring the transport layer
+- when dealing with memory allocators
+
+Time reference
+^^^^^^^^^^^^^^
+
+In order to operate in a time-based approach, the micro-ROS library will need at link time an implementation of the function `int clock_gettime(clockid_t, struct timespec *)` from the `POSIX specification <https://man7.org/linux/man-pages/man3/clock_gettime.3.html>`_ .
+
+This function will need to fill ``struct timespec *`` input argument implement with a monotonic time reference.
+In the case that the target platform does not provide this function, it is possible to implement it at application level and let the linker to resolve the symbol when linking the micro-ROS Client library.
+
+A reference implementation can be:
+
+.. code-block:: c
+
+  #include <sys/time.h>
+
+  #define USEC_IN_SEC 1000000
+  #define NSEC_IN_USEC 1000
+
+  int clock_gettime(clockid_t clock_id, struct timespec *tp)
+  {
+    (void) clock_id;
+
+    uint64_t microseconds_elapsed = my_platform_get_microseconds();
+
+    // Handle here possible rollovers of your platform timers if required.
+
+    tp->tv_sec = microseconds_elapsed / USEC_IN_SEC;
+    tp->tv_nsec = (microseconds_elapsed % USEC_IN_SEC) * NSEC_IN_USEC;
+
+    return 0;
+  }
+
+Transport layer
+^^^^^^^^^^^^^^^
+
+micro-ROS Client will need a transport implementation for communication with the micro-ROS Agent.
+
+Details on how to implement this transports can be found in :ref:`Custom Transports tutorial <tutorials_micro_custom_transports>`.
+
+.. note::
+
+    This tutorial explains how to implement custom transports in both micro-ROS Client and Agent side.
+    In the most common use case both parts are communicated using serial ports or UDP sockets.
+    This means that in the Client side, transport shall be provided using platform specific code at runtime, and in Agent side the built-in transports for ``serial`` and ``udp`` can be used.
```

Comment:
Maybe simplify this paragraph?

Reply:
49ebf86b29e214515890be611b1b2b456eb21cb0